ws-server/src/server.rs

```diff
@@ -120,6 +162,15 @@ impl Server {
 		self.root.register_subscription(subscribe_method_name, unsubscribe_method_name)
 	}
 
+	/// Register a new RPC subscription with the possibility to get the params in subscription request.
+	pub fn register_subscription_with_params<P: DeserializeOwned + Send + Sync + 'static>(
```

Comment:
Shouldn't this be the default? I don't see the point of having both `register_subscription` and `register_subscription_with_params` and force users to decide which they want to use. (But maybe it's like this just to experiment?)

Reply:
it's much cleaner otherwise we must have been a bunch of runtime checks not to send out responses on the wrong subscription when using send.

Like the user must still make the assumption whether the use `send_all` or `send based on this input to a specific subscription`, so why not have diffrent types/API for it? Then we can enforce more stuff at compile-time than at runtime

but it's still an experiment :)